cli/src/main/java/com/devonfw/tools/ide/tool/vscode/Vscode.java

```diff
@@ -28,4 +45,95 @@ protected String getBinaryName() {
     return "code";
   }
 
+  @Override
+  public void installPlugin(PluginDescriptor plugin) {
+
+  }
+
+  @Override
+  protected void installPlugins(Collection<PluginDescriptor> plugins) {
+
+    List<PluginDescriptor> pluginsToInstall = new ArrayList<>();
+    List<PluginDescriptor> pluginsToRecommend = new ArrayList<>();
+
+    for (PluginDescriptor plugin : plugins) {
+      if (plugin.isActive()) {
+        pluginsToInstall.add(plugin);
+      } else {
+        pluginsToRecommend.add(plugin);
+      }
+    }
+    doAddRecommendations(pluginsToRecommend);
+    doInstallPlugins(pluginsToInstall);
+
+  }
+
+  private void doInstallPlugins(List<PluginDescriptor> pluginsToInstall) {
+    if (pluginsToInstall.isEmpty()) {
+      this.context.info("No plugins to be installed");
+    } else {
+      for (PluginDescriptor plugin : pluginsToInstall) {
+        this.arguments.addValue("--install-extension");
+        this.arguments.addValue(plugin.getId());
+      }
+    }
+    runTool(ProcessMode.DEFAULT, null, this.arguments.asArray());
+  }
+
+  private void doAddRecommendations(List<PluginDescriptor> recommendations) {
+    Path extensionsJsonPath = this.context.getWorkspacePath().resolve(".vscode/extensions.json");
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    Map<String, Object> recommendationsMap;
+
+    try {
+      if (Files.exists(extensionsJsonPath) && Files.size(extensionsJsonPath) > 0) {
+        String content = Files.readString(extensionsJsonPath, StandardCharsets.UTF_8);
+        recommendationsMap = objectMapper.readValue(content, new TypeReference<Map<String, Object>>() {
```

Comment:
Use streaming instead of reading the entire `String` for efficiency:
```java
  try (BufferedReader reader = Files.newBufferedReader(extensionsJsonPath)) {
    recommendationsMap = objectMapper.readValue(reader, Map.class);
  }
```